files/models.py

```
@@ -1355,23 +1356,68 @@ def get_absolute_url(self):
+                        if new_path != input_path:
+                            os.rename(input_path, new_path)
+                            # Update the FileField to point to new path
+                            self.subtitle_file.name = self.subtitle_file.name.replace('.srt', '.vtt').replace('.SRT', '.vtt')
+                            self.save(update_fields=['subtitle_file'])
+                            logger.info(f"Renamed subtitle file from .srt to .vtt: {new_path}")
+                else:
+                    raise Exception("FFmpeg conversion failed - no output file created")
+                    
+            except Exception as e:
+                logger.error(f"Subtitle conversion failed for {input_path}: {str(e)}")
+                raise Exception(f"Could not convert SRT file to VTT format: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)